### PR TITLE
chore(frontend): Abstract the logic of retrieving pipeline definition in multi-doc yaml file.

### PR DESCRIPTION
**Pull request by jlyaoyuli**
Part of #9192

Add unit test to validate the helper function getPipelineDefFromYaml().

Add a multi-doc yaml for unit test.

### CODE REVIEW

**Review by chensun — APPROVED** [review]
/lgtm
/approve

Thanks!

File: frontend/src/lib/v2/WorkflowUtils.test.ts [hunk 1]
```diff
@@ -17,9 +17,15 @@ import { Workflow, WorkflowSpec, WorkflowStatus } from 'third_party/argo-ui/argo
...
 import { ComponentSpec } from 'src/generated/pipeline_spec';
 import fs from 'fs';
+import jsyaml from 'js-yaml';
 
-const V2_PIPELINESPEC_PATH = 'src/data/test/lightweight_python_functions_v2_pipeline_rev.yaml';
-const v2YamlTemplateString = fs.readFileSync(V2_PIPELINESPEC_PATH, 'utf8');
+const V2_LW_PIPELINESPEC_PATH = 'src/data/test/lightweight_python_functions_v2_pipeline_rev.yaml';
+const v2LWYamlTemplateString = fs.readFileSync(V2_LW_PIPELINESPEC_PATH, 'utf8');
+const V2_PVC_PIPELINESPEC_PATH = 'src/data/test/create_mount_delete_dynamic_pvc.yaml';
+const v2PVCYamlString = fs.readFileSync(V2_PVC_PIPELINESPEC_PATH, 'utf8');
```
Comment (zijianjoy) [inline frontend/src/lib/v2/WorkflowUtils.test.ts:25]: We should be consistent on the naming of const: It should be all capital in camel style.

File: frontend/src/lib/v2/WorkflowUtils.ts [hunk 2]
```diff
@@ -21,6 +21,18 @@ import { Workflow } from 'src/third_party/mlmd/argo_template';
...
+export function getPipelineDefFromYaml(template: string) {
+  // If pipeline_spec exists in the return value of safeload,
+  // which means the original yaml contains platform_spec,
+  // then the PipelineSpec(IR) is stored in 'pipeline_spec' field.
+  return jsyaml.safeLoad(template)[PIPELINE_SPEC_TEMPLATE_KEY] ?? jsyaml.safeLoad(template);
+}
+
+export function getPlatformDefFromYaml(template: string) {
+  return jsyaml.safeLoad(template)[PLATFORM_SPEC_TEMPLATE_KEY];
+}
```
Comment (zijianjoy) [inline frontend/src/lib/v2/WorkflowUtils.ts:35]: We might need to add some tests to validate return values `when kubernetes-platform exists` and `when no platform spec exists` of this exported function.

File: frontend/src/lib/v2/WorkflowUtils.ts [hunk 3]
```diff
@@ -21,6 +21,18 @@ import { Workflow } from 'src/third_party/mlmd/argo_template';
...
 // This key is used to retrieve the platform-agnostic pipeline definition
 export const PIPELINE_SPEC_TEMPLATE_KEY = 'pipeline_spec';
+export const PLATFORM_SPEC_TEMPLATE_KEY = 'platform_spec';
+
+export function getPipelineDefFromYaml(template: string) {
+  // If pipeline_spec exists in the return value of safeload,
+  // which means the original yaml contains platform_spec,
+  // then the PipelineSpec(IR) is stored in 'pipeline_spec' field.
+  return jsyaml.safeLoad(template)[PIPELINE_SPEC_TEMPLATE_KEY] ?? jsyaml.safeLoad(template);
+}
```
Comment (zijianjoy) [inline frontend/src/lib/v2/WorkflowUtils.ts:31]: We might need to add some tests to validate different scenarios of this exported function.

**Pull request merged**